### PR TITLE
Configuration

### PR DESCRIPTION
**Pull request by hedita**

### CODE REVIEW

File: src/components/Header/Header.scss [hunk 1]
```diff
@@ -7,3 +7,7 @@
   text-align: center;
   padding-bottom: $spacing-xxlarge;
 }
+
+.title.dark-mode {
+  color: white;
```
Comment (blacksrc) [inline src/components/Header/Header.scss]: Use variable instead.

File: src/components/Navbar/Navbar.jsx [hunk 2]
```diff
@@ -2,35 +2,61 @@ import React, { useContext } from "react";
...
+  const { darkMode } = useContext(DarkModeContext);
 
   return (
     <nav className="navbar">
       <ul className="filter-list">
         <li className="filter-item">
-          <Link className="navbar-item" to="/add">
+          <Link
+            className={
+              darkMode ? "navbar-item dark-mode" : "navbar-item light-mode"
```
Comment (blacksrc) [inline src/components/Navbar/Navbar.jsx]: Write this condition once and use the class here. Something like:
```js
const navbarItemClass = darkMode ? "navbar-item dark-mode" : "navbar-item light-mode";
```
and then use `navbarItemClass`.

File: src/components/Navbar/Navbar.scss [hunk 3]
```diff
@@ -22,13 +22,21 @@ ul {
 }
 
 .content {
-  background-color: $sixth-color;
+  // background-color: $sixth-color;
```
Comment (blacksrc) [inline src/components/Navbar/Navbar.scss]: ?

File: src/components/Navbar/Navbar.scss [hunk 4]
```diff
@@ -22,13 +22,21 @@ ul {
...
   border-radius: $spacing-xsmall;
   min-height: 95vh;
   margin-top: $spacing-small;
   padding: $spacing-small;
 }
 
+.dark-mode-bg {
+  background-color: black;
```
Comment (blacksrc) [inline src/components/Navbar/Navbar.scss]: Use variable.

File: src/components/Navbar/Navbar.scss [hunk 5]
```diff
@@ -47,3 +55,7 @@ ul {
   text-decoration: none;
   color: $primary-color;
 }
+
+.navbar-item.dark-mode {
+  color: white;
```
Comment (blacksrc) [inline src/components/Navbar/Navbar.scss]: Use variable.

File: src/components/TodoItem/TodoItem.jsx [hunk 6]
```diff
@@ -64,7 +66,10 @@ const TodoItem = ({ text, createdAt, taskId, getTasks, isDone }) => {
   return (
     <>
       {error && <p className="error-message">Something went wrong!</p>}
-      <li title={formatDate(createdAt)} className="todo-item">
+      <li
+        title={formatDate(createdAt)}
+        className={darkMode ? "todo-item dark-mode" : "todo-item light-mode"}
```
Comment (blacksrc) [inline src/components/TodoItem/TodoItem.jsx]:
```suggestion
        className={`todo-item {darkMode ? "dark-mode" : "light-mode"}`}
```

File: src/components/TodoItem/TodoItem.jsx [hunk 7]
```diff
@@ -85,7 +90,12 @@ const TodoItem = ({ text, createdAt, taskId, getTasks, isDone }) => {
             {text}
           </p>
         )}
-        <button className="delete-button" onClick={deleteTodo}>
+        <button
+          className={
+            darkMode ? "delete-button dark-mode" : "delete-button light-mode"
+          }
```
Comment (blacksrc) [inline src/components/TodoItem/TodoItem.jsx]: same as:
https://github.com/hedita/todo-react/pull/6/files#r1225368995

File: src/components/Header/Header.jsx [hunk 8]
```diff
@@ -1,8 +1,13 @@
...
+import React, { useContext } from "react";
 import "../Header/Header.scss";
+import { DarkModeContext } from "../../pages/ConfigurationPage/DarkModeContext";
 
 const Header = () => {
-  return <h1 className="title">Todo</h1>;
+  const { darkMode } = useContext(DarkModeContext);
+
+  return (
+    <h1 className={darkMode ? "title dark-mode" : "title light-mode"}>Todo</h1>
```
Comment (blacksrc) [inline src/components/Header/Header.jsx]: same as:
https://github.com/hedita/todo-react/pull/6/files#r1225368995

File: src/components/TodoItem/TodoItem.scss [hunk 9]
```diff
@@ -17,6 +17,11 @@
   margin-left: auto;
 }
 
+.delete-button.dark-mode {
+  color: white;
+  background-color: black;
```
Comment (blacksrc) [inline src/components/TodoItem/TodoItem.scss]: Use variables.

File: src/pages/ConfigurationPage/ConfigurationPage.jsx [hunk 10]
```diff
@@ -1,7 +1,14 @@
...
+import { useContext } from "react";
+import { DarkModeContext } from "./DarkModeContext";
 
 const ConfigurationPage = () => {
-  return <h2>configuration</h2>;
+  const { darkMode, toggleTheme } = useContext(DarkModeContext);
+
+  return (
+    <button className="lightSwitch" onClick={toggleTheme}>
+      {darkMode ? "light Mode" : "Dark Mode"}
```
Comment (blacksrc) [inline src/pages/ConfigurationPage/ConfigurationPage.jsx:13]: Pay attention to the details.
```suggestion
      {darkMode ? "Light Mode" : "Dark Mode"}
```

File: src/pages/ConfigurationPage/DarkModeContext.jsx [hunk 11]
```diff
@@ -0,0 +1,19 @@
...
+  const toggleTheme = () => {
+    setDarkMode(!darkMode);
+  };
+
+  return (
+    <DarkModeContext.Provider value={{ darkMode, toggleTheme }}>
+      {props.children}
+    </DarkModeContext.Provider>
+  );
+};
```
Comment (blacksrc) [inline src/pages/ConfigurationPage/DarkModeContext.jsx]: Please explain this in a voice.

File: src/pages/Main/Main.jsx [hunk 12]
```diff
@@ -0,0 +1,34 @@
...
+import Header from "../../components/Header/Header";
+import Navbar from "../../components/Navbar/Navbar";
+import { DarkModeContext } from "../ConfigurationPage/DarkModeContext";
+
+const Main = () => {
+  const { darkMode } = useContext(DarkModeContext);
+
+  return (
+    <main
+      className={darkMode ? "content dark-mode-bg" : "content light-mode-bg"}
```
Comment (blacksrc) [inline src/pages/Main/Main.jsx]: same as:
https://github.com/hedita/todo-react/pull/6/files#r1225368995

File: src/components/Navbar/Navbar.scss [hunk 13]
```diff
@@ -22,13 +22,21 @@ ul {
...
   min-height: 95vh;
   margin-top: $spacing-small;
   padding: $spacing-small;
 }
 
+.dark-mode-bg {
+  background-color: black;
+}
+
+.light-mode-bg {
```
Comment (blacksrc) [inline src/components/Navbar/Navbar.scss]: It's not used.

File: src/components/Navbar/Navbar.jsx (hunk 2 above)
Comment (blacksrc) [reply src/components/Navbar/Navbar.jsx]: I cannot see any class defined as `light-mode` in the css. What's the purpose of having that?

File: src/components/Navbar/Navbar.scss [hunk 14]
```diff
@@ -22,13 +22,21 @@ ul {
...
+  background-color: $sixth-color;
+}
+
 .navbar {
```
Comment (blacksrc) [inline src/components/Navbar/Navbar.scss:36]: You can use sass nested feature to accomplish this. Refactor the code in this way:
```scss
.box {
  /* base css properties with the light mode values */
  &.dark {
    /* override light mode properties */
  }
}
```

File: src/pages/ConfigurationPage/DarkModeContext.jsx (hunk 11 above)
Comment (hedita) [reply src/pages/ConfigurationPage/DarkModeContext.jsx]: Done.

File: src/components/Header/Header.scss (hunk 1 above)
Comment (hedita) [reply src/components/Header/Header.scss]: Done.

File: src/components/Navbar/Navbar.scss (hunk 3 above)
Comment (hedita) [reply src/components/Navbar/Navbar.scss]: Done.

File: src/components/Navbar/Navbar.scss (hunk 4 above)
Comment (hedita) [reply src/components/Navbar/Navbar.scss]: Done.

File: src/components/Navbar/Navbar.scss (hunk 5 above)
Comment (hedita) [reply src/components/Navbar/Navbar.scss]: Done.

File: src/components/TodoItem/TodoItem.scss (hunk 9 above)
Comment (hedita) [reply src/components/TodoItem/TodoItem.scss]: Done.

File: src/components/Navbar/Navbar.scss (hunk 13 above)
Comment (hedita) [reply src/components/Navbar/Navbar.scss]: Done.

File: src/components/Navbar/Navbar.scss (hunk 14 above)
Comment (hedita) [reply src/components/Navbar/Navbar.scss:36]: Done.

File: src/components/Navbar/Navbar.scss [hunk 15]
```diff
@@ -46,4 +50,8 @@ ul {
 .navbar-item {
   text-decoration: none;
   color: $primary-color;
+
+  &.navbar-item.dark-mode {
```
Comment (blacksrc) [inline src/components/Navbar/Navbar.scss]:
```suggestion
  &.dark-mode {
```

File: src/components/Navbar/Navbar.scss [hunk 16]
```diff
@@ -27,6 +27,10 @@ ul {
   min-height: 95vh;
   margin-top: $spacing-small;
   padding: $spacing-small;
+
+  &.dark-mode-bg {
```
Comment (blacksrc) [inline src/components/Navbar/Navbar.scss]:
```suggestion
  &.dark-mode {
```

File: src/components/TodoItem/TodoItem.scss [hunk 17]
```diff
@@ -15,6 +15,11 @@
   border: $spacing-xxxsmall solid $quaternary-color;
   border-radius: $spacing-xsmall;
   margin-left: auto;
+
+  &.delete-button.dark-mode {
```
Comment (blacksrc) [inline src/components/TodoItem/TodoItem.scss]:
```suggestion
  &.dark-mode {
```

File: src/components/TodoItem/TodoItem.jsx (hunk 7 above)
Comment (hedita) [reply src/components/TodoItem/TodoItem.jsx]: Done.

File: src/components/Header/Header.jsx (hunk 8 above)
Comment (hedita) [reply src/components/Header/Header.jsx]: Done.

File: src/pages/Main/Main.jsx (hunk 12 above)
Comment (hedita) [reply src/pages/Main/Main.jsx]: Done.

File: src/components/Navbar/Navbar.scss (hunk 15 above)
Comment (hedita) [reply src/components/Navbar/Navbar.scss]: Done.

File: src/components/Navbar/Navbar.scss (hunk 16 above)
Comment (hedita) [reply src/components/Navbar/Navbar.scss]: Done.

File: src/components/TodoItem/TodoItem.scss (hunk 17 above)
Comment (hedita) [reply src/components/TodoItem/TodoItem.scss]: Done.

File: src/components/Navbar/Navbar.jsx (hunk 2 above)
Comment (hedita) [reply src/components/Navbar/Navbar.jsx]: Done.

File: src/components/Navbar/Navbar.jsx [hunk 18]
```diff
@@ -6,47 +6,35 @@ import { DarkModeContext } from "../../pages/ConfigurationPage/DarkModeContext";
 const Navbar = () => {
   const { todosCount } = useContext(StatusLengthContext);
   const { darkMode } = useContext(DarkModeContext);
+  const navbarItemClass = darkMode
+    ? "navbar-item dark-mode"
+    : "navbar-item light-mode";
```
Comment (blacksrc) [inline src/components/Navbar/Navbar.jsx]:
```suggestion
const navbarItemClass = `navbar-item${darkMode ? " dark-mode" : ""}` 
```

Comment (hedita) [reply src/components/Navbar/Navbar.jsx]: Done.

Comment (hedita) [reply src/components/Navbar/Navbar.jsx]: Done.

File: src/components/Navbar/Navbar.jsx (hunk 2 above)
Comment (hedita) [reply src/components/Navbar/Navbar.jsx]: Done.

File: src/components/TodoItem/TodoItem.jsx [hunk 19]
```diff
@@ -64,7 +66,10 @@ const TodoItem = ({ text, createdAt, taskId, getTasks, isDone }) => {
   return (
     <>
       {error && <p className="error-message">Something went wrong!</p>}
```
Comment (blacksrc) [inline src/components/TodoItem/TodoItem.jsx]: Semantic.
```suggestion
      {error && <li className="error-message">Something went wrong!</li>}
```

File: src/components/Variables.scss [hunk 20]
```diff
@@ -28,6 +28,7 @@ $quaternary-color: #6c86ac;
 $fifth-color: #f1f3f4;
 $sixth-color: #ffffff;
 $seventh-color: green;
+$eighth-color: black;
```
Comment (blacksrc) [inline src/components/Variables.scss]:
```suggestion
$seventh-color: #00FF00;
$eighth-color: #000000;
```

File: src/pages/ConfigurationPage/DarkModeContext.jsx [hunk 21]
```diff
@@ -0,0 +1,26 @@
...
+const DarkModeContext = createContext(false);
+
+const DarkModeProvider = (props) => {
+  const [darkMode, setDarkMode] = useState(
+    () => JSON.parse(localStorage.getItem("darkMode")) || false
+  );
+
+  useEffect(() => {
+    localStorage.setItem("darkMode", JSON.stringify(darkMode));
+  }, [darkMode]);
```
Comment (blacksrc) [inline src/pages/ConfigurationPage/DarkModeContext.jsx:13]: Explain this in a voice. Line by line. function by function.

**Pull request merged**